ledger-service/http-json/src/test/scala/com/digitalasset/http/AbstractHttpServiceIntegrationTest.scala

```diff
@@ -604,6 +640,41 @@ abstract class AbstractHttpServiceIntegrationTest
       }: Future[Assertion]
   }
 
+  "create-and-exercise IOU_Transfer" in withHttpService { (uri, encoder, _) =>
+    import encoder.implicits._
+
+    val cmd: domain.CreateAndExerciseCommand[v.Record, v.Value] =
+      iouCreateAndExerciseTransferCommand()
+
+    val json: JsValue = SprayJson.encode2(cmd).valueOr(e => fail(e.shows))
+
+    postJsonRequest(uri.withPath(Uri.Path("/v1/create-and-exercise")), json)
+      .flatMap {
+        case (status, output) =>
+          status shouldBe StatusCodes.OK
+          inside(
+            decode2[domain.OkResponse, domain.ExerciseResponse[JsValue], Unit](output)
+          ) {
+            case \/-(response) =>
+              response.status shouldBe StatusCodes.OK
+              (response.warnings: Option[Unit]) shouldBe Option.empty[Unit]
+              inside(response.result.events) {
+                case List(
+                    domain.Contract(\/-(created0)),
+                    domain.Contract(-\/(archived0)),
+                    domain.Contract(\/-(created1))) =>
```

Comment:
created and archived right away + the final contract that is the result of the choice exercise.